ksqldb-engine/src/test/java/io/confluent/ksql/engine/KsqlEngineTest.java

```diff
@@ -1062,6 +1062,40 @@ public void shouldNotDeleteSchemaNorTopicForTable() throws Exception {
     assertThat(schemaRegistryClient.getAllSubjects(), hasItem("BAR-value"));
   }
 
+  @Test
+  public void shouldNotDeleteSchemaNorTopicForStream() throws Exception {
+    // Given:
+    givenTopicsExist("BAR");
+    final QueryMetadata query = KsqlEngineTestUtil.execute(
+        serviceContext,
+        ksqlEngine,
+        "create stream bar with (value_format = 'avro') as select * from test1;",
+        ksqlConfig, Collections.emptyMap()
+    ).get(0);
+
+    query.close();
+
+    final Schema schema = SchemaBuilder
+        .record("Test").fields()
+        .name("clientHash").type().fixed("MD5").size(16).noDefault()
+        .endRecord();
+
+    schemaRegistryClient.register("BAR-value", new AvroSchema(schema));
+
+    // When:
+    KsqlEngineTestUtil.execute(
+        serviceContext,
+        ksqlEngine,
+        "DROP STREAM bar;",
+        ksqlConfig,
+        Collections.emptyMap()
+    );
+
+    // Then:
+    assertThat(serviceContext.getTopicClient().isTopicExists("BAR"), equalTo(true));
```

Comment:
I know that you're just using the same format as the test above, but that test was written before `QueryCleanupService`, which I believe cleans up the topics asynchronously. To verify that this test would fail if something broke we should make sure that before executing the command the changelog topic subject was in SR and afterwards it is not.